File: mobile application development/front/react-native/src/api/tradesApi.ts
import { apiRequest } from './client';
import { API_PREFIX } from '../config/api';
import type { TradeRequest, TradeResponse } from './types';

export function buyTradeApi(body: TradeRequest) {
  return apiRequest<TradeResponse>(`${API_PREFIX}/trades/buy`, { method: 'POST', body });
}

export function sellTradeApi(body: TradeRequest) {
  return apiRequest<TradeResponse>(`${API_PREFIX}/trades/sell`, { method: 'POST', body });
}

File: mobile application development/front/kotlin/app/src/main/java/com/itmo/mybroker/api/TradesApi.kt
package com.itmo.mybroker.api

object TradesApi {
    private const val PREFIX = "/api/broker-app/v1"

    fun buy(body: TradeRequest): TradeResponse =
        ApiClient.request("$PREFIX/trades/buy", method = "POST", body = body)

    fun sell(body: TradeRequest): TradeResponse =
        ApiClient.request("$PREFIX/trades/sell", method = "POST", body = body)
}
